clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -111,12 +126,401 @@ class image_mem_wrapper {
+
+      res_width = desc.Width;
+      res_height = desc.Height;
+      res_depth = desc.Depth;
+      res_num_levels = (desc.MipLevels ? desc.MipLevels : 1);
+
+      channel = get_img_ch_info(desc.Format);
+      break;
+    }
+    default:
+      throw std::runtime_error("Unsupported DX11 resource type!");
+      break;
+    }
+
+    if (!_res_is_buffer) {
+      if (res_arr_size > 1) {
+        throw std::runtime_error(
+            "Images with multiple array levels are not supported. "
+            "Only standard and mipmap images are supported!");
+      }
+
+      _res_size_bytes = res_width * res_height * res_depth * res_arr_size *
+                        res_num_levels * channel.get_total_size();
+
+      sycl::ext::oneapi::experimental::image_type img_type =
+          (res_num_levels > 1)
+              ? sycl::ext::oneapi::experimental::image_type::mipmap
+              : sycl::ext::oneapi::experimental::image_type::standard;
+
+      _res_img_desc = sycl::ext::oneapi::experimental::image_descriptor{
+          sycl::range<3>{res_width, res_height, res_depth},
+#if (__SYCL_COMPILER_VERSION && __SYCL_COMPILER_VERSION < 20240527)
+          channel.get_channel_order(),
+#else
+          channel.get_channel_num(),
+#endif
+          channel.get_channel_type(),
+          img_type,
+          res_num_levels,
+          res_arr_size};
+    }
+  }
+
+  /// Helper function to create shared handle for DXD11 resource
+  HANDLE create_shared_handle(ID3D11Resource *resource) {
+    HANDLE win_nt_handle = nullptr;
+
+    IDXGIResource1 *pDXGIResource1;
+    resource->QueryInterface(__uuidof(IDXGIResource1),
+                             reinterpret_cast<void **>(&pDXGIResource1));
+
+    if (_res_is_buffer) {
+      pDXGIResource1->GetSharedHandle(&win_nt_handle);
+    } else {
+      pDXGIResource1->CreateSharedHandle(
+          nullptr, DXGI_SHARED_RESOURCE_READ | DXGI_SHARED_RESOURCE_WRITE,
+          ullptr, &win_nt_handle);
+    }
+
+    pDXGIResource1->Release();
+
+    return win_nt_handle;
+  }
+
+  /// Helper function to get the texture channel info of DX11 resource using
+  /// DXGI format
+  image_channel get_img_ch_info(DXGI_FORMAT format) {
+    image_channel channel;
+
+    switch (format) {
+    case DXGI_FORMAT_R16G16B16A16_FLOAT:
+      channel = image_channel::create<sycl::half4>;
+      break;
+    case DXGI_FORMAT_R8G8B8A8_UNORM:
+      channel = image_channel::create<sycl::uchar4>;
+      break;
+    case DXGI_FORMAT_R16G16_FLOAT:
+      channel = image_channel::create<sycl::half2>;
+      break;
+    case DXGI_FORMAT_R32_FLOAT:
+      channel = image_channel::create<sycl::float>;
+      break;
+    default:
+      throw std::runtime_error("Unsupported DX11 resource format!");
+      break;
+    }
+
+    return channel;
+  }
+};
+#else
+/// Stub interop_mem_wrapper implementation for non-Windows platforms
+class interop_mem_wrapper : public interop_mem_wrapper_base {
+public:
```

Comment:
Add `static_assert` for unsupported platforms.

Reply:
Addressed - no need to insert static_assert